File: core/databaseOps/src/main/scala/com/progressive/minds/chimera/core/databaseOps/modal/OrganizationHierarchy.scala
package com.progressive.minds.chimera.core.databaseOps.modal

import java.math.BigInteger
import java.sql.Timestamp

case class OrganizationHierarchy (
                                   private var org_id : BigInteger ,
                                   private var parent_org_id : String ,
                                   private var org_type_id : BigInteger,
                                   private var org_name : String,
                                   private var coo_owner : String,
                                   private var ops_lead : String,
                                   private var tech_lead : String,
                                   private var bus_owner : String,
                                   private var org_desc : String,
                                   private var org_email : String,
                                   private var org_ci : String,
                                   private var status : String = "Y",
                                   private var inserted_user : String = "System",
                                   private var insert_dt : Timestamp = new Timestamp(System.currentTimeMillis()),
                                   private var modified_by : String = "System",
                                   private var update_dt : Timestamp = new Timestamp(System.currentTimeMillis())
                            )
{
  // Getters
  def getOrgId : BigInteger  =  org_id
  def getParentOrgId : String  =  parent_org_id
  def getOrgTypeId : BigInteger  =  org_type_id
  def getOrgName : String  =  org_name
  def getCooOwner : String  =  coo_owner
  def getOpsLead : String  =  ops_lead
  def getTechLead : String  =  tech_lead
  def getBusOwner : String  =  bus_owner
  def getOrgDesc : String  =  org_desc
  def getOrgEmail : String  =  org_email
  def getOrgCi : String  =  org_ci
  def getStatus : String  =  status
  def getInsertedUser : String  =  inserted_user
  def getInsertDt : Timestamp  =  insert_dt
  def getModifiedBy : String  =  modified_by
  def getUpdateDt : Timestamp  =  update_dt


  // Setters
  def setOrgId(value: BigInteger) : Unit = {  org_id = value }
  def setParentOrgId(value: String) : Unit = {  parent_org_id = value }
  def setOrgTypeId(value: BigInteger) : Unit = {  org_type_id = value }
  def setOrgName(value: String) : Unit = {  org_name = value }
  def setCooOwner(value: String) : Unit = {  coo_owner = value }
  def setOpsLead(value: String) : Unit = {  ops_lead = value }
  def setTechLead(value: String) : Unit = {  tech_lead = value }
  def setBusOwner(value: String) : Unit = {  bus_owner = value }
  def setOrgDesc(value: String) : Unit = {  org_desc = value }
  def setOrgEmail(value: String) : Unit = {  org_email = value }
  def setOrgCi(value: String) : Unit = {  org_ci = value }
  def setStatus(value: String) : Unit = {  status = value }
  def setInsertedUser(value: String) : Unit = {  inserted_user = value }
  def setInsertDt(value: Timestamp) : Unit = {  insert_dt = value }
  def setModifiedBy(value: String) : Unit = {  modified_by = value }
  def setUpdateDt(value: Timestamp) : Unit = {  update_dt = value }
}

File: core/databaseOps/src/main/java/com/progressive/minds/chimera/core/databaseOps/model/metadata/updated/OrganizationHierarchy.java
package com.progressive.minds.chimera.core.databaseOps.model.metadata.updated;

import com.progressive.minds.chimera.core.databaseOps.annotation.Column;
import com.progressive.minds.chimera.core.databaseOps.annotation.Id;
import com.progressive.minds.chimera.core.databaseOps.annotation.Table;

import java.util.Date;
import java.util.UUID;

@Table(name = "organization_hierarchy")
public class OrganizationHierarchy {

    @Id
    @Column(name = "org_id")
    private UUID orgId;

    @Column(name ="parent_org_id")
    private UUID parentOrgId;

    @Column(name ="org_type_id")
    private UUID orgTypeId;

    @Column(name ="org_name")
    private String orgName;

    @Column(name ="coo_owner")
    private String cooOwner;

    @Column(name ="ops_lead")
    private String opsLead;

    @Column(name ="tech_lead")
    private String techLead;

    @Column(name ="bus_owner")
    private String busOwner;

    @Column(name ="org_desc")
    private String orgDesc;

    @Column(name ="org_email")
    private String orgEmail;

    @Column(name ="org_ci")
    private String orgCi;

    @Column(name ="user_field_1")
    private String userField1;

    @Column(name ="user_field_2")
    private String userField2;

    @Column(name ="user_field_3")
    private String userField3;

    @Column(name ="user_field_4")
    private String userField4;

    @Column(name ="user_field_5")
    private String userField5;

    @Column(name ="status")
    private String status;

    @Column(name ="inserted_user")
    private String insertedUser;

    @Column(name ="insert_dt")
    private Date insertDt;

    @Column(name ="modified_by")
    private String modifiedBy;

    @Column(name ="update_dt")
    private Date updateDt;

    // Getters and Setters

    public UUID getOrgId() {
        return orgId;
    }

    public void setOrgId(UUID orgId) {
        this.orgId = orgId;
    }

    public UUID getParentOrgId() {
        return parentOrgId;
    }

    public void setParentOrgId(UUID parentOrgId) {
        this.parentOrgId = parentOrgId;
    }

    public UUID getOrgTypeId() {
        return orgTypeId;
    }

    public void setOrgTypeId(UUID orgTypeId) {
        this.orgTypeId = orgTypeId;
    }

    public String getOrgName() {
        return orgName;
    }

    public void setOrgName(String orgName) {
        this.orgName = orgName;
    }

    public String getCooOwner() {
        return cooOwner;
    }

    public void setCooOwner(String cooOwner) {
        this.cooOwner = cooOwner;
    }

    public String getOpsLead() {
        return opsLead;
    }

    public void setOpsLead(String opsLead) {
        this.opsLead = opsLead;
    }

    public String getTechLead() {
        return techLead;
    }

    public void setTechLead(String techLead) {
        this.techLead = techLead;
    }

    public String getBusOwner() {
        return busOwner;
    }

    public void setBusOwner(String busOwner) {
        this.busOwner = busOwner;
    }

    public String getOrgDesc() {
        return orgDesc;
    }

    public void setOrgDesc(String orgDesc) {
        this.orgDesc = orgDesc;
    }

    public String getOrgEmail() {
        return orgEmail;
    }

    public void setOrgEmail(String orgEmail) {
        this.orgEmail = orgEmail;
    }

    public String getOrgCi() {
        return orgCi;
    }

    public void setOrgCi(String orgCi) {
        this.orgCi = orgCi;
    }

    public String getUserField1() {
        return userField1;
    }

    public void setUserField1(String userField1) {
        this.userField1 = userField1;
    }

    public String getUserField2() {
        return userField2;
    }

    public void setUserField2(String userField2) {
        this.userField2 = userField2;
    }

    public String getUserField3() {
        return userField3;
    }

    public void setUserField3(String userField3) {
        this.userField3 = userField3;
    }

    public String getUserField4() {
        return userField4;
    }

    public void setUserField4(String userField4) {
        this.userField4 = userField4;
    }

    public String getUserField5() {
        return userField5;
    }

    public void setUserField5(String userField5) {
        this.userField5 = userField5;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getInsertedUser() {
        return insertedUser;
    }

    public void setInsertedUser(String insertedUser) {
        this.insertedUser = insertedUser;
    }

    public Date getInsertDt() {
        return insertDt;
    }

    public void setInsertDt(Date insertDt) {
        this.insertDt = insertDt;
    }

    public String getModifiedBy() {
        return modifiedBy;
    }

    public void setModifiedBy(String modifiedBy) {
        this.modifiedBy = modifiedBy;
    }

    public Date getUpdateDt() {
        return updateDt;
    }

    public void setUpdateDt(Date updateDt) {
        this.updateDt = updateDt;
    }

    @Override
    public String toString() {
        return "OrganizationHierarchy{" +
                "orgId=" + orgId +
                ", parentOrgId=" + parentOrgId +
                ", orgTypeId=" + orgTypeId +
                ", orgName='" + orgName + '\'' +
                ", cooOwner='" + cooOwner + '\'' +
                ", opsLead='" + opsLead + '\'' +
                ", techLead='" + techLead + '\'' +
                ", busOwner='" + busOwner + '\'' +
                ", orgDesc='" + orgDesc + '\'' +
                ", orgEmail='" + orgEmail + '\'' +
                ", orgCi='" + orgCi + '\'' +
                ", userField1='" + userField1 + '\'' +
                ", userField2='" + userField2 + '\'' +
                ", userField3='" + userField3 + '\'' +
                ", userField4='" + userField4 + '\'' +
                ", userField5='" + userField5 + '\'' +
                ", status='" + status + '\'' +
                ", insertedUser='" + insertedUser + '\'' +
                ", insertDt=" + insertDt +
                ", modifiedBy='" + modifiedBy + '\'' +
                ", updateDt=" + updateDt +
                '}';
    }
}
